tests/Feature/MovieTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Tests\Feature;
+
+use Tests\TestCase;
+use App\Models\User;
+use App\Models\Movie;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class MovieTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /** @test */
+    public function an_admin_can_see_movies()
+    {
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->get('/movies')
+            ->assertOk();
+    }
+
+    /** @test */
+    public function an_admin_can_create_a_movie()
+    {
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->get('/movies/create')
+            ->assertOk();
+    }
+
+    /** @test */
+    public function a_non_admin_cannot_create_a_movie()
+    {
+        $user = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->actingAs($user)
+            ->get('/movies/create')
+            ->assertStatus(403);
+    }
+
+    /** @test */
+    public function an_admin_can_add_a_movie()
+    {
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->postJson('/movies', [
+                'title' => 'Sample Movie',
+                'synopsis' => 'This is a Sample Synopsis',
+                'year' => 2021,
+                'poster' => 'https://via.placeholder.com/600x750.png/00aa33?text=totam',
+                'trailer' => 'http://www.goyette.biz/',
+                'duration' => '190',
+            ])->assertOk();
+
+        $this->assertDatabaseHas('movies', [
+            'title' => 'Sample Movie',
+            'synopsis' => 'This is a Sample Synopsis',
+        ]);
+    }
+
+    /** @test */
+    public function an_admin_can_edit_a_movie()
+    {
+        $movie = Movie::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->get("/movies/{$movie->id}/edit")
+            ->assertOk();
+    }
+
+    /** @test */
+    public function a_non_admin_cannot_edit_a_movie()
+    {
+        $movie = Movie::factory()->create();
+        $user = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->actingAs($user)
+            ->get("/movies/{$movie->id}/edit")
+            ->assertStatus(403);
+    }
+
+    /** @test */
+    public function an_admin_can_update_a_movie()
+    {
+        $admin = User::factory()->admin()->create();
+        $movie = Movie::factory()->create();
```

Comment:
When testing update, the best approach is to create the movie with the values defined for each property. 

Then when acting as an admin you change the value to the properties and check in the database if the values have been updated. 

That way we know for sure that the values we send in the request have been saved in the DB.

The current test is using the factory, but we do not know the original value of the movie attributes, so the test feels incomplete. Then you are checking the DB for only two of the properties, which leaves space for errors to occur, like for example, how do we know that the trailer has been updated? 

😸